### ISSUE TITLE
uiNewHorizontalBox changes separator layout

### ISSUE THREAD

**Issue by Prefinem** [andlabs/libui#403]
This was originally found in libui-node, issue here: https://github.com/parro-it/libui-node/issues/122#issuecomment-406997366

Essentially, when you place a separator in a horizontal box, the separator will switch based on the size / aspect ratio of the window.

![screen shot 2018-07-22 at 4 52 20 pm](https://user-images.githubusercontent.com/2748262/43080849-48efd19a-8e56-11e8-8ac4-bf83e19c1c9a.png)
![screen shot 2018-07-22 at 4 52 27 pm](https://user-images.githubusercontent.com/2748262/43080857-4c1b9322-8e56-11e8-9795-139d156932fd.png)

**Comment by andlabs** [comment]
Yep, that's a bug, but I'm not sure how it would be fixed. NSSeparator determines its orientation based on its size, and your box is making it taller than it is wide. It's separators and boxes not playing well together, and I'm still not happy with my Auto Layout code to begin with (it's something I have to talk about in that one PR about fixed layouts)...

**Comment by mischnic** [comment]
> If you replace the separator with a slider, does the same thing happen?

No, then it works as expected: (I guess the top aligned label and other center aligned controls are a different issue)

![bildschirmfoto 2018-07-23 um 17 54 40](https://user-images.githubusercontent.com/4586894/43088023-7df9e9ca-8ea1-11e8-936e-4dfdb7158676.png)

**Comment by andlabs** [comment]
Keep open until fixed.
